### PR TITLE
Starting the process of multiple simulation

### PR DESCRIPTION
**Pull request by YohannParis**
**What**
- Display multiple models into the Simulation space.

**How**
- Use `selectedModelIds` to display multiple Models in the _Simulation_ view.
- The _Comparison_ view is broken, so instead, when multiple models are selected, a temporary _Open Simulation View_ button is displayed next to the _Comparison_ one. (c.f. the first screenshot)
- Move Store Simulation Types into the HMI type files
- Change the Simulation responsive grid to accept as many models as selected by the user. (This demands too much memory from the browser. Keep it under three models.)
- Update the Simulation store to handle multiple models.
- Update the Parameter panels to have a more natural input field.
- Disable the `Run` and `Save` button if some parameters have non numbers values.

**N.B.**
- Small fix on the EpiRendered to check a graph is available before highlighting nodes.
- Change the Parameters panel scale to `[-1, 1]` if the value is `0`.
- Added a _Loader_ on the model list and model page.
- Fix a small overflow issue on the _Global Graph_ component.
- For some reason, this is using a lot of memory, so if @mj3cheun or @adamocarolli can make sure I didn't shoot myself on the foot, that would be great.

**Screenshot**

Screen Shot 2021-07-26 at 13 54 56

**Video**

https://user-images.githubusercontent.com/636801/127382373-ed721c71-a1f0-49fa-9791-6fa3fe86d9ce.mov

### CODE REVIEW

File: client/src/views/Models/Model.vue [hunk 1]
```diff
@@ -13,7 +13,8 @@
       />
     </left-side-panel>
 
-    <section>
+    <loader v-if="!selectedGraph" loading="true" />
```
Comment (RosaRomeroGomez) [inline client/src/views/Models/Model.vue:16]: Nice, thank you.

File: client/src/views/Simulation/Simulation.vue [hunk 2]
```diff
@@ -145,43 +155,68 @@
...
+        } else if (this.setSelectedModels) {
+          gridMap.push([
+            'model_' + index,
+            'model-parameters-separator',
+            'parameters_' + index,
+            'parameters-variables-separator',
+            'variables_' + index,
+          ]);
+        }
+        // gridMap.push(['row-separator']);
```
Comment (RosaRomeroGomez) [inline client/src/views/Simulation/Simulation.vue:200]: Remove commented code.

**Comment by RosaRomeroGomez** [comment]
This is looking good. Thanks @YohannParis. Some comments below:

- It seems we are getting the scrolling bar even when we don't have parameters added to the panel, we might want to remove it or color it as the panel background so it is less noticeable. 
- Could we also make the separation between rows for the parameters panel grid smaller? In this way, we can fit more parameters without having to scroll much. 
image

File: client/src/store/modules/simulation.ts [hunk 3]
```diff
@@ -4,105 +4,118 @@ import * as Model from '@/types/typesModel';
...
-const state: SimulationState = {
+const state: HMI.SimulationState = {
   numberOfSavedRuns: 0,
-  parameters: [],
-  variables: [],
+  models: [],
+};
+
+const currentNumberOfRuns = (state: HMI.SimulationState): number => {
+  return state.models[0]?.parameters?.[0]?.values.length ?? 0;
```
[the remark this answers is not in the document]
Comment (YohannParis) [reply client/src/store/modules/simulation.ts:13]: This is correct, and it is a good idea to add a comment.

File: client/src/views/Simulation/Simulation.vue (hunk 2 above)
Comment (YohannParis) [reply client/src/views/Simulation/Simulation.vue:200]: This is still a draft, and there is a lot of bugs to iron out, this being one of them.

**Comment by YohannParis** [comment]
> This is looking good. Thanks @YohannParis. Some comments below:
> 
>     * It seems we are getting the scrolling bar even when we don't have parameters added to the panel, we might want to remove it or color it as the panel background so it is less noticeable.
> 
>     * Could we also make the separation between rows for the parameters panel grid smaller? In this way, we can fit more parameters without having to scroll much.

Screen Shot 2021-07-27 at 16 23 44

Both have been fixed.

File: client/src/services/DonuService.ts [hunk 4]
```diff
@@ -72,6 +71,7 @@ export const fetchDonuModels = async (): Promise<Model.Model[]> => {
           edges: model.graph.edges,
         },
       };
+      delete model.gromet;
```
Comment (RosaRomeroGomez) [inline client/src/services/DonuService.ts:74]: Why is this needed?

File: client/src/views/Simulation/components/ModelPanel.vue [hunk 5]
```diff
@@ -39,28 +45,53 @@
...
-    @Getter getSimParameters;
-    @Getter getSimVariables;
+    @Getter getSimModel;
+    @Getter getSelectedModelGraphType;
     @Action toggleParameter;
     @Action toggleVariable;
 
     settingsOpen: boolean = false;
 
     onNodeClick (selected: Graph.GraphNodeInterface): void {
```
Comment (RosaRomeroGomez) [inline client/src/views/Simulation/components/ModelPanel.vue:55]: I think this method should be probably called `onNodeDblClick` since it is not just a simple click, it gets triggered on double click.

File: client/src/views/Simulation/components/ModelPanel.vue [hunk 6]
```diff
@@ -39,28 +45,53 @@
...
+            .filter(node => highlightedLabels.includes(node.label))
+            .map(node => ({ id: node.id }));
+        }
+      }
+      return [];
     }
 
     get graph (): Graph.GraphInterface {
-      return this.model?.modelGraph?.[0]?.graph;
+      // return this.model?.modelGraph?.[0]?.graph;
```
Comment (RosaRomeroGomez) [inline client/src/views/Simulation/components/ModelPanel.vue:90]: Remove commented code.

**Review by RosaRomeroGomez — APPROVED** [review]
Great job @YohannParis

File: client/src/services/DonuService.ts (hunk 4 above)
Comment (mj3cheun) [reply client/src/services/DonuService.ts:123]: my understanding of how garbage collection works in `js` is that when the function finishes, the `model` object will be deleted by the garbage collector which includes the `gromet` property, so this deletion should not be necessary as its done automatically. I am not 100% certain though, I can test this out tomorrow by allocating like a 3gb buffer in the `model` object and see if the memory is released after the function is finished.

File: client/src/views/Simulation/Simulation.vue [hunk 7]
```diff
@@ -145,46 +159,83 @@
...
-      return this.expandedId
-      ? [[this.expandedId]]
-      : [
-          ['model', 'div1', 'parameters', 'div2', 'variables'],
-        ];
+      const gridMap: string[][] = [];
+
+      this.selectedModels.forEach(model => {
+        if (this.expandedId) {
+          gridMap.push([this.expandedId + '_' + model.id]);
```
Comment (mj3cheun) [inline client/src/views/Simulation/Simulation.vue:194]: coooool programmatically generated gridmaps

File: client/src/services/DonuService.ts (hunk 4 above)
Comment (mj3cheun) [reply client/src/services/DonuService.ts:123]: i did the buffer allocation test and i compared js heap size before and after adding `delete model.gromet`, doesnt look like this line is necessary

Comment (YohannParis) [reply client/src/services/DonuService.ts:123]: Awesome, thank you @mj3cheun
